provider/blob/src/static_data_provider.rs

```diff
@@ -103,21 +104,25 @@ impl BufferProvider for StaticDataProvider {
         let mut metadata = DataResponseMetadata::default();
         // TODO(#1109): Set metadata.data_langid correctly.
         metadata.buffer_format = Some(BufferFormat::Postcard1);
+        let idx = self
+            .data
+            .keys
+            .get0(&key.get_hash())
+            .ok_or(DataErrorKind::MissingResourceKey)
+            .and_then(|cursor| {
+                cursor
+                    .get1_copied_by(|bytes| req.options.strict_cmp(bytes).reverse())
+                    .ok_or(DataErrorKind::MissingResourceOptions)
+            })
+            .map_err(|kind| kind.with_req(key, req))?;
+        let bytes = self
+            .data
+            .buffers
+            .get(idx)
+            .ok_or_else(|| DataErrorKind::InvalidState.with_req(key, req))?;
```

Comment:
I don't like this `InvalidState` error, because it doesn't tell the client anything. Can we make it a custom error with a message or camouflage it as a postcard error (with a message)?

Reply:
I totally agree with you that `InvalidState` errors are not great because they don't tell the client anything, but I think your other suggestions are even worse (especially camouflaging it as a postcard error). I can attach a message string to the InvalidState error.

Reply:
I don't see how `Custom` with message is even worse than `InvalidState` with message. This is arguably not even invalid "state", it's an invalid input (the bytes buffer), which is why I suggested pretending that it's a postcard error. If we checked the invariant during deserialization it would become a postcard error, and for most invalid blobs it will be a postcard error (because the ZeroMap or its contents don't deserialize), but for a very small subset of invalid inputs we return `InvalidState` instead.

Reply:
It looks like the other place where we use `InvalidState` is in the `try_unwrap_owned` function.

> for a very small subset of invalid inputs we return `InvalidState` instead.

I'm convinced by this argument. With postcard, we do

```rust
        crate::DataError::custom("Postcard deserialize").with_display_context(&e)
```

so we can use `custom` here as well.

Reply:
Re `try_unwrap_owned`, I'd be happy to use `custom` there and remove the `InvalidState` error kind.